lib/mpool/src/mclass.c

```diff
@@ -383,6 +386,18 @@ mcid_to_mclass(enum mclass_id mcid)
     return HSE_MCLASS_INVALID;
 }
 
+void
+mclass_io_ops_set(enum hse_mclass mclass, struct io_ops *io)
+{
+    INVARIANT(io);
+
+#ifdef HAVE_PMEM
+    *io = (mclass == HSE_MCLASS_PMEM) ? io_pmem_ops : io_sync_ops;
```

Comment:
Does libpmem also work on non-persistent memory? What if you have libpmem installed, but you have no persistent-memory available?

Reply:
Persistent memory can be configured only in a pmem media class. So, if libpmem is installed and there's no persistent memory available then there can't be a pmem media class. The IO path will use the sync backend in this case.

Reply:
Seems like we could consolidate media class ID and media class. They are just one off from eachother. Anyway to MCID_INVALID to u8 max and capacity et al down 1? Or potentially increment enum hse_mclass to start at 1?